src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -0,0 +1,336 @@
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+public class LocalMqtt5Client implements MessageClient<MqttMessage> {
+    private static final Logger LOGGER = LogManager.getLogger(LocalMqtt5Client.class);
+    public static final String TOPIC = "topic";
+    private static final int MIN_WAIT_RETRY_IN_SECONDS = 1;
+    private static final int MAX_WAIT_RETRY_IN_SECONDS = 120;
+    private static final QOS QOS = software.amazon.awssdk.crt.mqtt5.QOS.AT_LEAST_ONCE;
+    private final URI brokerUri;
+    private final String clientId;
+    private Mqtt5Client client;
+    private Consumer<MqttMessage> messageHandler;
+    private final MQTTClientKeyStore mqttClientKeyStore;
+    private final ExecutorService executorService;
+
+    @Getter(AccessLevel.PROTECTED)
+    private Set<String> subscribedLocalMqttTopics = ConcurrentHashMap.newKeySet();
+    private Set<String> toSubscribeLocalMqttTopics = new HashSet<>();
+
+    private final RetryUtils.RetryConfig mqttExceptionRetryConfig =
+            RetryUtils.RetryConfig.builder().initialRetryInterval(Duration.ofSeconds(1L))
+                    .maxRetryInterval(Duration.ofSeconds(120L)).maxAttempt(Integer.MAX_VALUE)
+                    .retryableExceptions(Collections.singletonList(MqttException.class)).build();
+
+    private final Object subscribeLock = new Object();
+    CompletableFuture<SubAckPacket> subscribeFuture;
+
+    private final AtomicReference<CompletableFuture<Void>> stopFuture = new AtomicReference<>(null);
+    @Getter(AccessLevel.PACKAGE)
+    private final Mqtt5ClientOptions.LifecycleEvents connectionEventCallback =
+            new Mqtt5ClientOptions.LifecycleEvents() {
+        @Override
+        public void onAttemptingConnect(Mqtt5Client client,
+                                       OnAttemptingConnectReturn onAttemptingConnectReturn) {
+            LOGGER.atDebug().log("Attempting to connect to AWS IoT Core");
+        }
+
+        @Override
+        public void onConnectionSuccess(Mqtt5Client client, OnConnectionSuccessReturn onConnectionSuccessReturn) {
+
+        }
+
+        @Override
+        public void onConnectionFailure(Mqtt5Client client, OnConnectionFailureReturn onConnectionFailureReturn) {
+
+        }
+
+        @Override
+        public void onDisconnection(Mqtt5Client client, OnDisconnectionReturn onDisconnectionReturn) {
+
+        }
+
+        @Override
+        public void onStopped(Mqtt5Client client, OnStoppedReturn onStoppedReturn) {
+            client.close();
+            CompletableFuture<Void> f = stopFuture.get();
+            if (f != null) {
+                f.complete(null);
+            }
+        }
+    };
+
+    /**
+     * TODO: Implement constructor
+     */
+    public LocalMqtt5Client(@NonNull URI brokerUri, @NonNull String clientId, MQTTClientKeyStore mqttClientKeyStore,
+                             ExecutorService executorService, Mqtt5ClientOptions mqtt5ClientOptions) {
```

Comment:
i don't think we want to expose `Mqtt5ClientOptions`, what would it look like if we just passed in what we needed and then have LocalMqtt5Client be responsible for creating options?

Reply:
on `Mqtt5ClientOptions`, we'll need to set fields like host/port, etc